src/App.tsx

```diff
@@ -1,61 +1,142 @@
+import { useState } from 'react';
 import './App.scss';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList/TodoList';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+export function getUserFromId(userId: number): User | null {
+  return usersFromServer.find((person) => person.id === userId) || null;
+}
+
+export const todosWithUser: Todo[] = todosFromServer.map((todo) => ({
+  ...todo,
+  user: getUserFromId(todo.userId),
+}));
 
 export const App = () => {
+  const [todos, setTodos] = useState(todosWithUser);
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+  const [isTitleError, setIsTitleError] = useState(false);
+  const [userIdError, setUserIdError] = useState(false);
+  const userList = usersFromServer;
+  const highestTodoId = Math.max(...todos.map(todo => (todo.id)));
+
+  function resetForm() {
+    setUserId(0);
+    setTitle('');
+  }
+
+  function changeTitle(event: React.ChangeEvent<HTMLInputElement>) {
+    // I opted to do the optional instruction
+    const legalCharacters = /^[\wа-яА-Яa-zA-Z ]+/;
+    const croppedInput = event.target.value.split('').filter((char) => (
+      legalCharacters.test(char)
+    )).join('');
+
+    setTitle(croppedInput);
+
+    if (isTitleError && croppedInput.length > 0) {
```

Comment:
If it's string you don't need to check for it's length
```suggestion
    if (isTitleError && croppedInput) {
```